peer/brontide_test.go

```diff
@@ -862,6 +860,114 @@ func TestCustomShutdownScript(t *testing.T) {
 	}
 }
 
+// TestStaticRemoteDowngrade tests that we downgrade our static remote feature
+// bit to optional if we have legacy channels with a peer. This ensures that
+// we can stay connected to peers that don't support the feature bit that we
+// have channels with.
+func TestStaticRemoteDowngrade(t *testing.T) {
+	t.Parallel()
+
+	var (
+		// We set the same legacy feature bits for all tests, since these are
+		// not relevant to our test scenario
+		rawLegacy = lnwire.NewRawFeatureVector(lnwire.AnchorsOptional)
```

Comment:
Not sur if really matters for this test, but could use the `SingleFunderBit` here instead, since in realuty `AnchorsOptional` implies `StaticRemoteKeyOptional`.